### PR TITLE
【Hackathon No.5】为 Paddle 新增 bucketize

### PR DESCRIPTION
**Pull request by Li-fAngyU**
PaddlePaddle Hackathon 第三期 No.5 为 Paddle 新增 bucketize

### CODE REVIEW

File: rfcs/APIs/20220708_api_design_for_bucketize.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+## API实现方案
+主要按下列步骤进行实现,实现位置为`paddle/tensor/math.py`与`searchsorted`方法放在一起：
+1. 使用`len(sorted_sequence)`检验参数`sorted_sequence`的维度。
+2. 使用`paddle.searchsorted`得到输入的桶索引。
+
+
+# 六、测试和验收的考量
+测试考虑的case如下：
+
+- 和pytorch结果的数值的一致性, `paddle.bucketize`,和`torch.bucketize`结果是否一致；
```
Comment (jeff41404) [inline rfcs/APIs/20220708_api_design_for_bucketize.md:113]: unit test can only use numpy

File: rfcs/APIs/20220708_api_design_for_bucketize.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+# 六、测试和验收的考量
+测试考虑的case如下：
+
+- 和numpy结果的数值的一致性, `paddle.bucketize`,和`numpy.searchsorted`结果是否一致；
+- 参数`right`为True和False时输出的正确性；
+- `out_int32`为True和False时输出dtype正确性；
+- 未输入`right`时的输出正确性；
+- 未输入`out_int32`时的输出正确性；
+- 错误检查：输入`x`不是Tensor时,能否正确抛出错误；
+- 错误检查：`axis`所指维度在当前Tensor中不合法时能正确抛出错误。
```
Comment (jeff41404) [inline rfcs/APIs/20220708_api_design_for_bucketize.md:119]: there is no parameter of `axis`, why do we need this case

**Comment by Li-fAngyU** [comment]
Done

**Review by jeff41404 — APPROVED** [review]
LGTM

**Pull request merged**